spec/api-menu-spec.js

```diff
@@ -118,14 +121,36 @@ describe('Menu module', () => {
             position: 'endof=numbers'
           }
         ])
-        assert.equal(menu.items[0].id, 'numbers')
-        assert.equal(menu.items[1].label, '1')
-        assert.equal(menu.items[2].label, '2')
-        assert.equal(menu.items[3].label, '3')
-        assert.equal(menu.items[4].id, 'letters')
-        assert.equal(menu.items[5].label, 'a')
-        assert.equal(menu.items[6].label, 'b')
-        assert.equal(menu.items[7].label, 'c')
+
+        assert.equal(menu.items[1].id, 'numbers')
+        assert.equal(menu.items[2].label, '1')
+        assert.equal(menu.items[3].label, '2')
+        assert.equal(menu.items[4].label, '3')
+        assert.equal(menu.items[5].id, 'letters')
+        assert.equal(menu.items[6].label, 'a')
+        assert.equal(menu.items[7].label, 'b')
+        assert.equal(menu.items[8].label, 'c')
+      })
+
+      it('should filter excess menu separators', () => {
+        const menu = Menu.buildFromTemplate([
+          {
+            type: 'separator'
+          }, {
+            label: 'a'
+          }, {
+            label: 'b'
+          }, {
+            label: 'c'
+          }, {
+            type: 'separator'
+          }
+        ])
+
+        assert.equal(menu.items.length, 3)
+        assert.equal(menu.items[0].label, 'a')
+        assert.equal(menu.items[1].label, 'b')
+        assert.equal(menu.items[2].label, 'c')
```

Comment:
Another good edge case to test for would be [ sep, sep, a, b, c, sep, sep ] -> [ a, b, c ]